src/Agent.Worker/TestResults/Parser.cs

```diff
@@ -0,0 +1,120 @@
+﻿using Microsoft.TeamFoundation.TestClient.PublishTestResults;
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Microsoft.VisualStudio.Services.Agent.Worker.TestResults
+{
+    public interface IParser : IExtension
```

Comment:
In General, now if someone wants to add there own parser, they need to do at two places. I think one in agent will be like proxy kind of stuff. 

So was thinking do we need to add a Factory in the PublishTestResutls, based on the Name it will give us the right parser and on that we can call the Parse.?

Reply:
Yeah that would be better, else we will always need to add in 2 places.